lightcrafts/src/main/java/com/lightcrafts/model/ImageEditor/ZoneFinder.java

```diff
@@ -62,6 +75,11 @@ public void addNotify() {
     @Override
     public void removeNotify() {
         // This method gets called when this Preview is removed.
+        stopSegmenter();
+        if (debounceTimer != null) {
+            debounceTimer.stop();
+            debounceTimer = null;
+        }
         super.removeNotify();
     }
```

Comment:
To improve readability, let's also create stopDebounceTimer() method and use it here.

```suggestion
        // This method gets called when this Preview is removed.
        stopSegmenter();
        stopDebounceTimer();
        super.removeNotify();
    }

    private void stopDebounceTimer() {
        if (debounceTimer != null) {
            debounceTimer.stop();
            debounceTimer = null;
        }
    }
```